vfs.go

```diff
@@ -462,3 +465,112 @@ func (v *VFS) NumOfFragmentsInPath(path string) (int, error) {
 
 	return numOfFragmentsData.NumOfFolders, nil
 }
+
+// Close a file. This is flushes the buffered data into the file when the file
+// was opened in write (or append) mode. It is particularly important to be
+// called after S3 writes, as otherwise the writes will not take effect.
+func (v *VFSfh) Close() error {
+
+	ret := C.tiledb_vfs_close(v.context.tiledbContext, v.tiledbVFSfh)
+
+	if ret != C.TILEDB_OK {
+		return fmt.Errorf("Unknown error in VFS.Close: %s", v.context.LastError())
+	}
+
+	v.Free()
+	return nil
+}
+
+// Read part of a file
+func (v *VFSfh) Read(p []byte) (int, error) {
```

Comment:
Read and Write signatures end up the same and this confuses me. I don't know if the out type argument is used in go, but i 'd prefer to allocate memory in this function taking the number of bytes needed and returning the []byte

Reply:
We can't edit the Read/Write signatures they are part of the interface(s): https://golang.org/pkg/io/#Reader